revolv/static/js/base/dashboard_project.js

```diff
@@ -74,32 +93,68 @@ function DashboardCircles() {
         var dimension = (2 * radius) + (2 * padding);
         var translateVar = (radius + padding) * 0.5;
 
-        var svg = d3.select(".dashboard-data-section-current .internal-graphics-container");
+        var svg = d3.select(this.generateSelector(htmlClassNames["internalGraphicsContainer"]));
         // we have this check whether it is null because the currently selected project might not be completed
         if (!svg.empty()) {
             svg = svg.attr("width", dimension)
                 .attr("height", dimension)
-                .append("g");
+                .append("g").attr("class", htmlClassNames["badgeGroupingContainer"].substring(1));
 
             var stroke = radius * 0.2;
-            var circleGrouping = svg.append("g").attr("class", "repayment-badge-circle-grouping").attr("stroke-width", stroke + "px");
-            var partialGrouping = svg.append("g").attr("class", "repayment-badge-partial-grouping").attr("stroke-width", stroke + "px");
+            var circleGrouping = svg.append("g").attr("class", htmlClassNames["badgeCircleGrouping"].substring(1)).attr("stroke-width", stroke + "px");
+            var partialGrouping = svg.append("g").attr("class", htmlClassNames["badgePartialGrouping"].substring(1)).attr("stroke-width", stroke + "px");
 
-            drawD3PartialCircle(circleGrouping, ["repayment-badge-circle"], radius, padding, 1);
-            drawD3PartialCircle(partialGrouping, ["repayment-badge-line"], radius, padding, partialCompleteness);
+            drawD3PartialCircle(circleGrouping, [htmlClassNames["badgeCircle"].substring(1)], radius, padding, 1);
```

Comment:
I think it would be better to store the "un-dotted" names in the htmlClassNames dict and prepend the dot wherever needed rather than store the "dotted" version and take the substring wherever the dot is *not* needed. That way, the values you store aren't tied to the fact that they're used as selectors as much